tests/unit/transformers/test_utils.py

```diff
@@ -222,41 +222,66 @@ def test_try_convert_to_dtype():
     pd.testing.assert_series_equal(output_convertibe, expected_data_convertibe)
 
 
-@patch('rdt.transformers.utils.LOGGER')
-def test_learn_rounding_digits_more_than_15_decimals(logger_mock):
-    """Test the learn_rounding_digits method with more than 15 decimals.
-
-    If the data has more than 15 decimals, return None and raise warning.
-    """
+def test_learn_rounding_digits():
+    """Test it for various cases."""
     # Setup
-    data = pd.Series(np.random.random(size=10).round(20), name='col')
-
-    # Run
-    output = learn_rounding_digits(data)
-
-    # Assert
-    logger_msg = "No rounding scheme detected for column '%s'. Data will not be rounded."
-    logger_mock.info.assert_called_once_with(logger_msg, 'col')
-    assert output is None
-
-
-def test_learn_rounding_digits_less_than_15_decimals():
-    """Test the learn_rounding_digits method with less than 15 decimals.
-
-    If the data has less than 15 decimals, the maximum number of decimals
-    should be returned.
-
-    Input:
-    - An array that contains floats with a maximum of 3 decimals and a
-        NaN.
-    Output:
-    - 3
-    """
-    data = pd.Series(np.array([10, 0.0, 0.1, 0.12, 0.123, np.nan]))
-
-    output = learn_rounding_digits(data)
+    test_cases = [
+        # data_1: Basic decimal places test
+        (pd.Series([10, 0.0, 0.1, 0.12, 0.123, np.nan]), 3),
+        # data_2: Large numbers with decimals
+        (
+            pd.Series([
+                1234567890123456.7,
+                12345678901234567.89,
+                123456789012345678.901,
+            ]),
+            None,
+        ),
+        # data_3: Consistent single decimal place
+        (
+            pd.Series([
+                1.1,
+                11.1,
+                111.1,
+                1111.1,
+                11111.1,
+                111111.1,
+                1111111.1,
+                11111111.1,
+                111111111.1,
+                1111111111.1,
+                11111111111.1,
+                111111111111.1,
+                1111111111111.1,
+                11111111111111.1,
+            ]),
+            1,
+        ),
+        # data_4-8: Various precision tests
+        (pd.Series([123456789012345.6789]), None),
+        (pd.Series([12345678901.234]), 3),
+        (pd.Series([12345678901.2345]), 4),
+        (pd.Series([0.1234567890123456]), None),
+        (pd.Series([0.123456789012345]), 15),
+        # data_9-10: Integer tests
+        (pd.Series([123456789012345]), 0),
```

Comment:
I think that you are trying to set up multiple tests cases in here, just use `@pytest.mark.parametrize(`, it is better since it will run one test per case and not having to 'wait' for all them. The positive part of it it is that will run all of them even if one fails, meanwhile yours will stop on the first `assertion` error.